pkg/test/metadata.go

```diff
@@ -44,10 +44,15 @@ func RandomName() string {
 }
 
 func ObjectMeta(overrides ...metav1.ObjectMeta) metav1.ObjectMeta {
-	return MustMerge(metav1.ObjectMeta{
-		Name:      RandomName(),
+	return MustMerge(ObjectMetaWithoutNamespace(metav1.ObjectMeta{
 		Namespace: "default",
-		Labels:    map[string]string{DiscoveryLabel: "unspecified"}, // For cleanup discovery
+	}), overrides...)
+}
+
+func ObjectMetaWithoutNamespace(overrides ...metav1.ObjectMeta) metav1.ObjectMeta {
```

Comment:
Thoughts on inverting this? 
- NamespacedObjectMeta()
- ObjectMeta()

It requires a couple of changes in the `test` pkg, but is conceptually more straight forward.

Reply:
Fixed
https://github.com/aws/karpenter-core/pull/89/commits/bb00c2bb2ba6a2ed04ff3a80304b78a1fe4d544d